src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx

```diff
@@ -0,0 +1,229 @@
+import { useEffect, useState } from 'react';
+import { IFormValues } from 'app/core/types';
+import { Listbox } from '@headlessui/react';
+import { CaretDown, Check, ArrowLeft } from '@phosphor-icons/react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import Button from 'app/shared/components/Button/Button';
+import Avatar from 'app/shared/components/Avatar';
+import BaseCheckbox from 'app/shared/components/forms/BaseCheckbox/BaseCheckbox';
+import { useTranslationContext } from 'app/i18n/provider/TranslationProvider';
+import './ShareInviteDialog.scss';
+
+interface ShareInviteDialog {
+  onInviteUser: () => void;
+}
+
+interface UsersToInvite {
+  email: string;
+  userRole: string;
+}
+
+const ShareInviteDialog = (props: ShareInviteDialog): JSX.Element => {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    watch,
+    resetField,
+  } = useForm<IFormValues>({ mode: 'onChange' });
+  const { translate } = useTranslationContext();
+  const emailRegex =
+    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+  const email = watch('email');
+  const [userRole, setUserRole] = useState<string>('editor');
+  const [usersToInvite, setUsersToInvite] = useState<Array<UsersToInvite>>([]);
+  const [notifyUser, setNotifyUser] = useState<boolean>(false);
+  const [messageText, setMessageText] = useState<string>('');
+  const [inviteUsersButton, setInviteUsersButton] = useState<boolean>(true);
```

Comment:
I think it is a bit confusing this name, because if `inviteUserButton` is `true` it means that the invite button is disabled. Maybe `isInviteButtonDisabled` is more descriptive with what it does.

Reply:
Done